src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AvoidDuplicateElementInitialization.cs

```diff
@@ -0,0 +1,142 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftQualityGuidelinesAnalyzersResources.AvoidDuplicateElementInitializationDescription), MicrosoftQualityGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftQualityGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: true,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly SyntaxGenerator s_generator = SyntaxGenerator.GetGenerator(new AdhocWorkspace(), LanguageNames.CSharp);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeOperation, OperationKind.ObjectCreation);
+        }
+
+        private static void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var objectInitializer = (IObjectCreationOperation)context.Operation;
+            if (objectInitializer.Initializer == null)
+            {
+                return;
+            }
+
+            var initializedElementIndexes = new HashSet<object[]>(ConstantArgumentEqualityComparer.Instance);
+
+            foreach (var initializer in objectInitializer.Initializer.Initializers)
+            {
+                if (initializer is ISimpleAssignmentOperation assignment &&
+                    assignment.Target is IPropertyReferenceOperation propertyReference &&
+                    propertyReference.Arguments.Length != 0)
+                {
+                    var values = GetConstantArgumentValues(propertyReference.Arguments);
+                    if (values != null && !initializedElementIndexes.Add(values))
+                    {
+                        var indexesText = string.Join(", ", values.Select(value => s_generator.LiteralExpression(value)));
```

Comment:
Why do we need to generate a LiteralExpression here? Can't we just use value.ToString()?

Reply:
The only problem I see there is if the index arguments are string literals, the error message would show the actual string value instead of the quoted string literal. However, I see now that I didn't update this part to be language agnostic earlier. I'll update it now, but let me know if you thing `value.ToString()` is sufficient and I'll remove the generators.